config/shared_config.go

```diff
@@ -192,15 +200,56 @@ func loadSharedConfig(ctx context.Context, configs configs) (Config, error) {
 		profile = defaultSharedConfigProfile
 	}
 
-	files, ok, err = getSharedConfigFiles(ctx, configs)
+	configurationFiles, ok, err = getSharedConfigFiles(ctx, configs)
 	if err != nil {
 		return nil, err
 	}
 	if !ok {
-		files = DefaultSharedConfigFiles
+		configurationFiles = DefaultSharedConfigFiles
 	}
 
-	return NewSharedConfig(profile, files)
+	credentialsFiles, ok, err = getSharedCredentialsFiles(ctx, configs)
+	if err != nil {
+		return nil, err
+	}
+	if !ok {
+		credentialsFiles = DefaultSharedCredentialsFiles
+	}
+
+	// setup logger if log configuration warning is seti
+	var logger logging.Logger
+	logWarnings, found, err := getLogConfigurationWarnings(ctx, configs)
+	if err != nil {
+		return SharedConfig{}, err
+	}
+	if found && logWarnings {
+		logger, found, err = getLogger(ctx, configs)
+		if err != nil {
+			return SharedConfig{}, err
+		}
+		if !found {
+			logger = logging.NewStandardLogger(os.Stderr)
+		}
+	}
+
+	return NewSharedConfig(ctx, profile, sharedConfigsLoadInfo{
```

Comment:
Input parameter not exported, `sharedConfigsLoadInfo`.  Consider changing `sharedConfigsLoadInfo` to functional inputs, e.g. `LoadSharedConfigProfileOptions`, and change `NewSharedConfig` to `LoadSharedConfigProfile`

In addition, the `CredentialsFiles` and `ConfigFiles` are optional parameters and `credentialsFiles = DefaultSharedCredentialsFiles` is done internally to the `LoadSharedConfigProfile` function if not specified.